internal/linker/link_node_modules.bzl

```diff
@@ -122,6 +122,34 @@ def get_module_mappings(label, attrs, vars, rule_kind, srcs = [], workspace_name
     elif label.workspace_root:
         mr = "%s/%s" % (label.workspace_root, mr)
 
+    # Consider the module_root attribute when resolving mappings to allow for ts_libraries compiled with it set.
+
+    if attrs.module_root and attrs.module_root != ".":
+        mr = "%s/%s" % (mr, attrs.module_root)
+        if attrs.module_root.endswith(".ts"):
+            if workspace_name:
+                mr = mr.replace(".d.ts", "")
+
+            # Validate that sources are underneath the module root.
+            # module_roots ending in .ts are a special case, they are used to
+            # restrict what's exported from a build rule, e.g. only exports from a
+            # specific index.d.ts file. For those, not every source must be under the
+            # given module root.
+
+        else:
+            for s in srcs:
+                short_path = s.short_path
+
+                # Execroot paths for external repositories should start with external/
+                # But the short_path property of file gives the relative path from our workspace
+                # instead. We must correct this to compare with the module_root which is an
+                # execroot path.
+                if short_path.startswith("../"):
+                    short_path = "external/" + short_path[3:]
+                if not short_path.startswith(mr):
+                    fail(("all sources must be under module root: %s, but found: %s" %
+                          (mr, short_path)))
```

Comment:
this is currently lifted straight from here: https://github.com/bazelbuild/rules_typescript/blob/0ab3a13ab610ad033a3c1ebd8a2080f7f251b624/internal/common/module_mappings.bzl#L79-L107

Reply:
I think we want to drop this on purpose - new rollup bundle just uses npm mechanisms to resolve things so it has to be a `package.json#main` instead